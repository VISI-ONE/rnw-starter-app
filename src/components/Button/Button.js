import React from 'react';
import { string, func, number, object, oneOfType } from 'prop-types';
import { Text, TouchableHighlight } from 'react-native';

const Button = ({
  style,
  onClick,
  text,
  textStyle
}) => (
  <TouchableHighlight onPress={onClick} style={style} >
    <Text style={textStyle}>{text}</Text>
  </TouchableHighlight>
);

Button.propTypes = {
  style: oneOfType([object, string, number]),
  textStyle: oneOfType([object, string, number]),
  onClick: func,
  text: string
};

export default Button;
