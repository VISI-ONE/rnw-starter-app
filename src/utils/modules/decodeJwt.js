/* eslint-disable no-cond-assign, no-bitwise, no-mixed-operators, no-plusplus */
// from https://stackoverflow.com/questions/42829838/react-native-atob-btoa-not-working-without-remote-js-debugging#42833475
const chars = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/=';
const decode = (input = '') => {
  const str = input.replace(/=+$/, '');
  let output = '';

  if (str.length % 4 === 1) {
    // failed to decode
    return null;
  }
  for (let bc = 0, bs = 0, buffer, i = 0;
    buffer = str.charAt(i++);

    ~buffer && (bs = bc % 4 ? bs * 64 + buffer : buffer,
    bc++ % 4) ? output += String.fromCharCode(255 & bs >> (-2 * bc & 6)) : 0
  ) {
    buffer = chars.indexOf(buffer);
  }

  return output;
};

const decodeJwt = (token) => {
    const base64Url = token.split('.')[1];
    const base64 = base64Url.replace('-', '+').replace('_', '/');
    return JSON.parse(decode(base64));
};

export default decodeJwt;
