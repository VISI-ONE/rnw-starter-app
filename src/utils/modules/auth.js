import { AsyncStorage } from 'react-native';
import decodeJwt from './decodeJwt';
import { isDev } from '../common';
const TOKEN_KEY = 'authtoken';

const request = (url, method, body, extraHeaders) => {
	let headers = {
		'Content-Type': 'application/json'
	};

	if (extraHeaders) {
		headers = { ...headers, ...extraHeaders };
	}

	const options = {
    method,
    headers,
  };

	if (method === 'POST') {
		options.body = body;
	}
	return fetch(url, options)
  .then(res => res.json());
};

export const logout = () => {
	AsyncStorage.removeItem(TOKEN_KEY);
};

export const login = (username, password) => {
  return request('some-url', 'POST', JSON.stringify({ username, password }))
	.then(res => {
		if (res && res.status === 'success' && res.token) {
			setUserToken(res.token);
		}
		return res;
	})
  .catch(err => err);
};

export const isTokenExpired = (token) => {
	try {
		const decodedToken = decodeJwt(token);
		const tokenExperationTimestamp = parseInt(decodedToken.exp, 10); // eg 1505727610 // seconds since UTC 1970.. .

		const dateNow = new Date();
		const clientTimestamp = Math.round(dateNow.getTime() / 1000);

		if (tokenExperationTimestamp < clientTimestamp) {
			// TokenExperation is in the past (less then Now)
			return true;
		}
		// TokenExperation is in the future (more then Now)
		return false;
	} catch (e) {
		return true;
	}
};

export const getUserToken = () => {
	return AsyncStorage.getItem(TOKEN_KEY)
	.then(token => {
		if (token && !isTokenExpired(token)) {
			return token;
		}
		return undefined;
	});
};

export const setUserToken = (token) => {
	if (token) {
		try {
			AsyncStorage.setItem(TOKEN_KEY, token);
		} catch (e) {} // eslint-disable-line no-empty
	}
};

export const checkUserToken = () => {
	return getUserToken()
	.then(token => {
		if (token) {
			return request('refresh url', 'GET', null, { Bearer: token })
			.then(res => {
				if (res && res.status === 'success' && res.newToken) {
					setUserToken(res.newToken);
				}
				return res;
			})
		  .catch(err => err);
		}
		return null;
	});
};
