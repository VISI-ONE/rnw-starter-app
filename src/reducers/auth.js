import { LOGIN, LOGOUT } from '../actions/types';
import { logout } from '../utils/modules/auth';

const initialState = {
  user: null,
  isLoggedIn: false
};

export default (state = initialState, action) => {
  switch (action.type) {
    case LOGIN:
      return {
        ...state,
        user: action.user,
        isLoggedIn: true
      };
    case LOGOUT:
      logout();
      return {
        ...state,
        user: null,
        isLoggedIn: false
      };
    default:
      return state;
  }
};
