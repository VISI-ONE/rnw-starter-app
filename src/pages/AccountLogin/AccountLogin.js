import React, { Component } from 'react';
import { connect } from 'react-redux';
import { StyleSheet, Text, TextInput, ScrollView, View } from 'react-native';
import { login as loginAction } from '../../actions/auth';
import Button from '../../components/Button';
import Routing from '../../utils/routing';
import { isWeb, extractValueFromEvent, extractKeyFromEvent } from '../../utils/common';

const { Redirect } = Routing;

export class AccountLogin extends Component {
  state = {
    email: '',
    password: '',
    error: '',
  }

  onEmailChange = (event) => {
    this.setState({ email: extractValueFromEvent(event) });
  }
  onPassChange = (event) => {
    this.setState({ password: extractValueFromEvent(event) });
  }

  submit = () => {
    const { dispatchLogin } = this.props;
    const { email } = this.state;

    this.setState({ error: '' });
    dispatchLogin({ email });
  }

  handleKeyDown = (event) => {
    if (extractKeyFromEvent(event) === 'Enter') {
        this.submit();
    }
  }

  render() {
    const { error } = this.state;
    const { isLoggedIn, translations } = this.props;

    if (isLoggedIn) {
      return <Redirect to="/" />;
    }

    const content = (
      <View style={styles.loginForm}>
        <Text style={styles.label}>{translations.login__email__label}</Text>
        <TextInput
          onChangeText={this.onEmailChange}
          placeholder={translations.login__email__placeholder}
          style={styles.textinput}
          onKeyPress={this.handleKeyDown}
        />
        <Text style={styles.label}>{translations.login__password__label}</Text>
        <TextInput
          secureTextEntry
          onChangeText={this.onPassChange}
          placeholder={translations.login__password__placeholder}
          style={styles.textinput}
          onKeyPress={this.handleKeyDown}
        />
        <Button style={styles.button} textStyle={styles.buttonText} onClick={this.submit} text={translations.login} />
        {!!error && <Text>{error}</Text>}
      </View>
    );

    if (isWeb) {
      return content;
    }
    return (
      <ScrollView>
        {content}
      </ScrollView>
    );
  }
}

export const styles = StyleSheet.create({
  loginForm: {
    flex: 1,
    backgroundColor: '#F5FCFF',
    flexDirection: 'column',
  },
  button: {
    margin: 10,
    alignSelf: 'center',
  },
  buttonText: {
    color: '#00F',
  },
  label: {
    color: '#000',
    margin: 10,
  },
  textinput: {
    backgroundColor: '#fff',
    borderWidth: 1,
    color: '#000',
    padding: 5,
    margin: 10,
  },
  error: {
    color: '#f00',
    margin: 10,
  },
});

export const mapStateToProps = state => ({
  isLoggedIn: state.auth.isLoggedIn,
  translations: {
    login__email__label: state.locale.translations.login__email__label,
    login__password__label: state.locale.translations.login__password__label,
    login__email__placeholder: state.locale.translations.login__email__placeholder,
    login__password__placeholder: state.locale.translations.login__password__placeholder,
    login: state.locale.translations.login,
  }
});

export const mapDispatchToProps = dispatch => ({
  dispatchLogin: user => dispatch(loginAction(user)),
});

export default connect(mapStateToProps, mapDispatchToProps)(AccountLogin);
