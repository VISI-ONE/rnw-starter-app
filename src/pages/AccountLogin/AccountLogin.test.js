/* eslint-disable no-plusplus, no-return-assign */
import { ScrollView, Text, Platform, TextInput, View } from 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { styles, mapStateToProps, mapDispatchToProps, AccountLogin } from './AccountLogin';
import {
  login as loginAction
} from '../../actions/auth';
import Button from '../../components/Button/Button';
import translations from '../../utils/locales/en-US';
import Routing from '../../utils/routing';
import Router from '../../__mocks__/Router';
import { login as loginRequest, resetCounter, getCounter } from '../../utils/modules/auth';
import { isWeb } from '../../utils/common';

const { Redirect } = Routing;

const fakeUserData = {
  email: 'fakeUser1',
  password: 123456
};

jest.mock('../../utils/modules/auth', () => {
  let checkUserTokenCallCounter = 0;
  const token = '007';
  return {
    resetCounter: () => checkUserTokenCallCounter = 0,
    getCounter: () => checkUserTokenCallCounter,
    login: () => {
      checkUserTokenCallCounter++;
      if (global.shouldReturnResponseForAccountLogin === 'error') {
        return Promise.reject(new Error('error'));
      }
      if (global.shouldReturnResponseForAccountLogin) {
        return Promise.resolve({ status: 'success', token });
      }
      return Promise.resolve({ message: 'error' });
    }
  };
});

describe('AccountLogin page', () => {
  describe('rendering', () => {
    it('render properly - ' + Platform.OS, () => {
      const login = jest.fn();
      const reactTree = renderer.create(<AccountLogin isLoggedIn={false} dispatchLogin={login} translations={translations} />);
      expect(reactTree.toJSON()).toMatchSnapshot();
    });

    it('has the right components and structure', () => {
      const login = jest.fn();
      const enzymeTree = shallow(<AccountLogin isLoggedIn={false} dispatchLogin={login} translations={translations} />);
      const content = (
        <View style={styles.loginForm}>
          <Text style={styles.label}>{translations.login__email__label}</Text>
          <TextInput onChangeText={this.onEmailChange} placeholder={translations.login__email__placeholder} style={styles.textinput} />
          <Text style={styles.label}>{translations.login__password__label}</Text>
          <TextInput
            secureTextEntry
            onChangeText={this.onPassChange}
            placeholder={translations.login__password__placeholder}
            style={styles.textinput}
          />
          <Button style={styles.button} textStyle={styles.buttonText} onClick={this.login} text={translations.login} />
        </View>
      );
      const wrapper = shallow(
        Platform.OS === 'web' ? content : (
          <ScrollView>
            {content}
          </ScrollView>
        )
      );

      expect(enzymeTree.html()).toEqual(wrapper.html());
    });

    it('renders an error when has one in the state', () => {
      const login = jest.fn();
      const error = 'error';
      const tree = shallow(<AccountLogin isLoggedIn={false} dispatchLogin={login} translations={translations} />);
      const RenderedError = shallow(<Text>{error}</Text>);
      tree.setState({ email: '', password: '', error });
      const errorNode = tree.find({ children: error });
      expect(errorNode.length).toEqual(1);
      expect(errorNode.html()).toEqual(RenderedError.html());
    });
  });

  describe('functionality', () => {
    it('submit function should submits the form and calls the login action', () => {
      global.shouldReturnResponseForAccountLogin = true;
      resetCounter();
      const login = jest.fn();
      const tree = shallow(<AccountLogin isLoggedIn={false} dispatchLogin={login} translations={translations} />);
      const btn = tree.find({ text: translations.login });

      tree.setState(fakeUserData);
      btn.simulate('click');
      expect(getCounter()).toEqual(1);
      return loginRequest().then(() => {
        expect(login).toBeCalledWith({});
        global.shouldReturnResponseForAccountLogin = false;
      });
    });

    it('redirects to home when user is logged in', () => {
      const login = jest.fn();
      const tree = renderer.create(
        <Router>
          <AccountLogin isLoggedIn dispatchLogin={login} translations={translations} />
        </Router>
      ).root;

      expect(tree.findByType(Redirect).props.to).toEqual('/');
    });

    it('onEmailChange should store the email in the state', () => {
      const login = jest.fn();
      const value = 'test';
      const instance = renderer.create(
        <AccountLogin dispatchLogin={login} translations={translations} />
      ).getInstance();
      instance.onEmailChange(value);

      expect(instance.state).toEqual({ email: value, password: '', error: '' });
    });

    it('onPassChange should store the password in the state', () => {
      const login = jest.fn();
      const value = 'test';
      const instance = renderer.create(
        <AccountLogin dispatchLogin={login} translations={translations} />
      ).getInstance();
      instance.onPassChange(value);

      expect(instance.state).toEqual({ email: '', password: value, error: '' });
    });

    it('submit should set an error when no token returned', () => {
      global.shouldReturnResponseForAccountLogin = false;
      const login = jest.fn();
      const tree = shallow(<AccountLogin isLoggedIn={false} dispatchLogin={login} translations={translations} />);
      const btn = tree.find({ text: translations.login });

      btn.simulate('click');
      return loginRequest().then(() => {
        expect(tree.state()).toEqual({ email: '', password: '', error: 'error' });
      });
    });

    it('submit should set an error when response sends an error', () => {
      global.shouldReturnResponseForAccountLogin = 'error';
      const login = jest.fn();
      resetCounter();
      const treeInstance = renderer.create(<AccountLogin isLoggedIn={false} dispatchLogin={login} translations={translations} />).getInstance();
      treeInstance.setState = jest.fn();
      treeInstance.submit();
      expect(getCounter()).toEqual(1);
      global.shouldReturnResponseForAccountLogin = false;

      return loginRequest().then(() => {
        return loginRequest().then(() => {
          expect(treeInstance.setState.mock.calls[1]).toEqual([{ error: 'error' }]);
        });
      });
    });

    it('handleKeyDown calls should not call submit when receives any other key except for Enter', () => {
      const login = jest.fn();
      const key = 'Enter';
      const instance = renderer.create(
        <AccountLogin dispatchLogin={login} translations={translations} />
      ).getInstance();
      instance.submit = jest.fn();

      instance.handleKeyDown(isWeb ? { key: 'Up' } : { nativeEvent: { key: 'Up' } });
      expect(instance.submit).not.toBeCalled();
      expect(instance.submit.mock.calls.length).toEqual(0);

      instance.handleKeyDown(isWeb ? { key: 'Down' } : { nativeEvent: { key: 'Down' } });
      expect(instance.submit).not.toBeCalled();
      expect(instance.submit.mock.calls.length).toEqual(0);
      instance.handleKeyDown(isWeb ? { key: '42' } : { nativeEvent: { key: '42' } });
      expect(instance.submit).not.toBeCalled();
      expect(instance.submit.mock.calls.length).toEqual(0);

      instance.handleKeyDown(isWeb ? { key } : { nativeEvent: { key } });

      expect(instance.submit).toBeCalled();
      expect(instance.submit.mock.calls.length).toEqual(1);
    });
  });

  describe('redux', () => {
    it('mapStateToProps', () => {
      const fakeState = { auth: { isLoggedIn: false }, locale: { translations } };
      const props = mapStateToProps(fakeState);
      expect(props).toEqual({
        isLoggedIn: false,
        translations: {
          login__email__label: translations.login__email__label,
          login__password__label: translations.login__password__label,
          login__email__placeholder: translations.login__email__placeholder,
          login__password__placeholder: translations.login__password__placeholder,
          login: translations.login,
        },
      });
    });

    it('mapDispatchToProps should return the functions that would create the proper actions ', () => {
      const mockDispatch = jest.fn();
      const props = mapDispatchToProps(mockDispatch);
      expect(Object.keys(props)).toEqual(['dispatchLogin']);
      expect(typeof props.dispatchLogin).toEqual('function');
      props.dispatchLogin(fakeUserData);

      expect(mockDispatch.mock.calls.length).toEqual(1);
      expect(mockDispatch).toBeCalledWith(loginAction(fakeUserData));
    });
  });
});
