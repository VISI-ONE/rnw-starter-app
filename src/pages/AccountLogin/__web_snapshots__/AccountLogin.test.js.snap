// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountLogin page rendering render properly - web 1`] = `
<div
  className="rn-alignItems-1awozwy rn-alignSelf-1pz39u2 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flex-13awgt0 rn-flexGrow-1m1wadx rn-flexShrink-1awmn5t rn-flexDirection-18u37iz rn-height-1pi2tsx rn-justifyContent-1777fci rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
>
  <div
    className="rn-alignItems-1oszu61 rn-alignSelf-1pz39u2 rn-backgroundColor-44z8sh rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flex-13awgt0 rn-flexGrow-1m1wadx rn-flexShrink-1awmn5t rn-flexDirection-eqz5dr rn-height-1pi2tsx rn-justifyContent-1777fci rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
  >
    <div
      className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-15b02fe rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-1b43r93 rn-marginTop-156q2ks rn-marginRight-zso239 rn-marginBottom-15d164r rn-marginLeft-1n0xq6e rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
      dir="auto"
    >
      Email
    </div>
    <input
      autoCapitalize="sentences"
      autoComplete="on"
      autoCorrect="on"
      className="rn-appearance-30o5oe rn-backgroundColor-44z8sh rn-borderTopColor-kqr9px rn-borderRightColor-q0dj5p rn-borderBottomColor-1ah7hsa rn-borderLeftColor-137uh4u rn-borderTopLeftRadius-ou6ah9 rn-borderTopRightRadius-t12b5v rn-borderBottomRightRadius-zmljjp rn-borderBottomLeftRadius-pm2fo rn-borderTopWidth-5kkj8d rn-borderRightWidth-13l2t4g rn-borderBottomWidth-qklmqi rn-borderLeftWidth-1ljd8xs rn-boxSizing-deolkf rn-color-15b02fe rn-font-1lw9tu2 rn-marginTop-156q2ks rn-marginRight-zso239 rn-marginBottom-15d164r rn-marginLeft-1n0xq6e rn-paddingTop-glunga rn-paddingRight-19gegkz rn-paddingBottom-1inuy60 rn-paddingLeft-qhyqy2 rn-resize-1dz5y72"
      dir="auto"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onSelect={[Function]}
      placeholder="example@example.com"
      readOnly={false}
      type="text"
    />
    <div
      className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-15b02fe rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-1b43r93 rn-marginTop-156q2ks rn-marginRight-zso239 rn-marginBottom-15d164r rn-marginLeft-1n0xq6e rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
      dir="auto"
    >
      Password
    </div>
    <input
      autoCapitalize="sentences"
      autoComplete="on"
      autoCorrect="on"
      className="rn-appearance-30o5oe rn-backgroundColor-44z8sh rn-borderTopColor-kqr9px rn-borderRightColor-q0dj5p rn-borderBottomColor-1ah7hsa rn-borderLeftColor-137uh4u rn-borderTopLeftRadius-ou6ah9 rn-borderTopRightRadius-t12b5v rn-borderBottomRightRadius-zmljjp rn-borderBottomLeftRadius-pm2fo rn-borderTopWidth-5kkj8d rn-borderRightWidth-13l2t4g rn-borderBottomWidth-qklmqi rn-borderLeftWidth-1ljd8xs rn-boxSizing-deolkf rn-color-15b02fe rn-font-1lw9tu2 rn-marginTop-156q2ks rn-marginRight-zso239 rn-marginBottom-15d164r rn-marginLeft-1n0xq6e rn-paddingTop-glunga rn-paddingRight-19gegkz rn-paddingBottom-1inuy60 rn-paddingLeft-qhyqy2 rn-resize-1dz5y72"
      dir="auto"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyPress={[Function]}
      onSelect={[Function]}
      placeholder="******"
      readOnly={false}
      type="password"
    />
    <div
      className="rn-alignItems-1oszu61 rn-alignSelf-1kihuf0 rn-backgroundColor-wib322 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-cursor-1loqt21 rn-display-6koalj rn-flexShrink-1pxmb3b rn-flexBasis-7vfszb rn-flexDirection-eqz5dr rn-marginTop-156q2ks rn-marginRight-zso239 rn-marginBottom-15d164r rn-marginLeft-1n0xq6e rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-touchAction-1gvxusu rn-userSelect-lrvibr rn-zIndex-1lgpqti"
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      tabIndex="0"
    >
      <div
        className="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-759hku rn-display-1471scf rn-font-1lw9tu2 rn-fontFamily-10u92zi rn-fontSize-1b43r93 rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
        dir="auto"
      >
        Login
      </div>
    </div>
  </div>
</div>
`;
